src/Compilers/CSharp/Test/Emit/Emit/CompilationEmitTests.cs

```diff
@@ -857,6 +857,55 @@ private void CompareAssemblies(string sourceTemplate, string change1, string cha
         }
 
         [Fact]
+        public void RefAssemblyClient_RefReadonlyParameters()
+        {
+            VerifyRefAssemblyClient(@"
+public class C
+{
+    public void RR_input(ref readonly int x) => throw null;
+    public ref readonly int RR_output() => throw null;
+    public ref readonly int P => throw null;
+    public ref readonly int this[ref readonly int i] => throw null;
+    public delegate ref readonly int Delegate(ref readonly int i);
+}
+public static class Extensions
+{
+    public static void RR_extension(ref readonly this int x) => throw null;
+    public static void R_extension(ref this int x) => throw null;
+}",
+@"class D
+{
+    void M(C c, ref readonly int y)
+    {
+        c.RR_input(y);
+        VerifyRR(c.RR_output());
+        VerifyRR(c.P);
+        VerifyRR(c[y]);
+        C.Delegate x = VerifyDelegate;
+        y.RR_extension();
+        1.RR_extension();
+        y.R_extension(); // error 1
+        1.R_extension(); // error 2
+    }
+    void VerifyRR(ref readonly int y) => throw null;
+    ref readonly int VerifyDelegate(ref readonly int y) => throw null;
+}",
+comp => comp.VerifyDiagnostics(
+                // (9,24): error CS0123: No overload for 'VerifyDelegate' matches delegate 'C.Delegate'
+                //         C.Delegate x = VerifyDelegate;
+                Diagnostic(ErrorCode.ERR_MethDelegateMismatch, "VerifyDelegate").WithArguments("VerifyDelegate", "C.Delegate").WithLocation(9, 24),
+                // (12,9): error CS8329: Cannot use variable 'ref readonly int' as a ref or out value because it is a readonly variable
+                //         y.R_extension(); // error 1
+                Diagnostic(ErrorCode.ERR_RefReadonlyNotField, "y").WithArguments("variable", "ref readonly int").WithLocation(12, 9),
+                // (13,9): error CS1510: A ref or out value must be an assignable variable
+                //         1.R_extension(); // error 2
+                Diagnostic(ErrorCode.ERR_RefLvalueExpected, "1").WithLocation(13, 9)
+                ));
+
+            // TODO REVIEW what am I doing wrong with delegate scenario?
```

Comment:
My guess is that the "IsReadOnlyAttribute" is missing in the ref assembly. 
Scenarios that are more permissive - like `ref` variable (it is ref because `readonly` was lost) is still valid to pass to an `in` parameter. So those scenarios work. But when RefKind must match exactly (like in delegate conversions), it fails.